crates/bevy_feathers/src/controls/checkbox.rs

```diff
@@ -49,74 +46,61 @@ struct CheckboxOutline;
-pub fn checkbox<C: SpawnableList<ChildOf> + Send + Sync + 'static, B: Bundle>(
-    props: CheckboxProps,
-    overrides: B,
-    label: C,
-) -> impl Bundle {
-    (
+pub fn checkbox(props: CheckboxProps) -> impl Scene {
+    bsn! {
         Node {
             display: Display::Flex,
             flex_direction: FlexDirection::Row,
             justify_content: JustifyContent::Start,
             align_items: AlignItems::Center,
             column_gap: Val::Px(4.0),
-            ..Default::default()
-        },
+        }
         CoreCheckbox {
-            on_change: props.on_change,
-        },
-        CheckboxFrame,
-        Hovered::default(),
-        CursorIcon::System(bevy_window::SystemCursorIcon::Pointer),
-        TabIndex(0),
-        ThemeFontColor(tokens::CHECKBOX_TEXT),
+            on_change: {props.on_change.clone()},
+        }
+        CheckboxFrame
+        Hovered
+        // TODO: port CursorIcon to GetTemplate
+        // CursorIcon::System(bevy_window::SystemCursorIcon::Pointer)
+        TabIndex(0)
+        ThemeFontColor(tokens::CHECKBOX_TEXT)
         InheritableFont {
-            font: HandleOrPath::Path(fonts::REGULAR.to_owned()),
+            font: fonts::REGULAR,
             font_size: 14.0,
-        },
-        overrides,
-        Children::spawn((
-            Spawn((
+        }
+        [(
+            Node {
+                width: size::CHECKBOX_SIZE,
+                height: size::CHECKBOX_SIZE,
+                border: UiRect::all(Val::Px(2.0)),
+            }
+            CheckboxOutline
+            BorderRadius::all(Val::Px(4.0))
+            ThemeBackgroundColor(tokens::CHECKBOX_BG)
+            ThemeBorderColor(tokens::CHECKBOX_BORDER)
+            [(
+                // Cheesy checkmark: rotated node with L-shaped border.
                 Node {
-                    width: size::CHECKBOX_SIZE,
-                    height: size::CHECKBOX_SIZE,
-                    border: UiRect::all(Val::Px(2.0)),
-                    ..Default::default()
-                },
-                CheckboxOutline,
-                BorderRadius::all(Val::Px(4.0)),
-                ThemeBackgroundColor(tokens::CHECKBOX_BG),
-                ThemeBorderColor(tokens::CHECKBOX_BORDER),
-                children![(
-                    // Cheesy checkmark: rotated node with L-shaped border.
-                    Node {
-                        position_type: PositionType::Absolute,
-                        left: Val::Px(4.0),
-                        top: Val::Px(0.0),
-                        width: Val::Px(6.),
-                        height: Val::Px(11.),
-                        border: UiRect {
-                            bottom: Val::Px(2.0),
-                            right: Val::Px(2.0),
-                            ..Default::default()
-                        },
-                        ..Default::default()
+                    position_type: PositionType::Absolute,
+                    left: Val::Px(4.0),
```

Comment:
I'd really like to be able to say `4.0px` here, or even `4px`.

Reply:
I'd like to include `use Val::{Px, Percent}` in the bevy_ui prelude, which would allow the still rust-ey but much shorter: `left: Px(4.0)` syntax.

Reply:
to clarify, can we have bsn convert `Px(4)` to `Px(4.into())` or `Px(f32::from(4))`